db/document.go

```diff
@@ -694,15 +694,14 @@ func (doc *document) UnmarshalJSON(data []byte) error {
 	root := documentRoot{SyncData: &syncData{History: make(RevTree)}}
 	err := json.Unmarshal([]byte(data), &root)
 	if err != nil {
-		base.Warnf(base.KeyAll, "Error unmarshaling doc %q: %s", base.UD(doc.ID), err)
-		return pkgerrors.Wrapf(err, "Failed to UnmarshalJSON() doc with id: %s", doc.ID)
+		return pkgerrors.WithStack(base.RedactErrorf("Failed to UnmarshalJSON() doc with id: %s.  Error: %v", base.UD(doc.ID), err))
```

Comment:
Again, are we OK with dropping the warning log here?

Reply:
This one is harder to trace, but I'd give the same reasoning as on the other case .. and push the responsibility up to callers.